### PR TITLE
Add --no-cache-dir option to pip install in Dockerfile

### PR DESCRIPTION
**Pull request by bro-sin**
This commit adds the `--no-cache-dir` option to the `pip install` command in the [`Dockerfile`](https://github.com/nerfstudio-project/nerfstudio/blob/main/Dockerfile).  By including this option, the cache during Python package installation is disabled, resulting in a significant reduction (approximately 3.2GB) in disk usage during `docker build`.

Without this option, the Docker image would contain the following cache, which is unnecessary:
```shell
du -sh /home/user/.cache/pip
# 3.2G    /home/user/.cache/pip
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -96,7 +96,7 @@ RUN git clone --branch 3.8 https://github.com/colmap/colmap.git --single-branch
     mkdir build && \
     cd build && \
     cmake .. -DCUDA_ENABLED=ON \
-             -DCMAKE_CUDA_ARCHITECTURES=${CUDA_ARCHITECTURES} && \
```
Comment (jb-ye) [inline Dockerfile:99]: fix indent

File: Dockerfile [hunk 2]
```diff
@@ -119,44 +119,44 @@ ENV PATH="${PATH}:/home/user/.local/bin"
...
 
 # Upgrade pip and install packages.
-RUN python3.10 -m pip install --upgrade pip setuptools pathtools promise pybind11
+RUN python3.10 -m pip install --no-cache-dir --upgrade pip setuptools pathtools promise pybind11
 # Install pytorch and submodules
-RUN CUDA_VER=${CUDA_VERSION%.*} && CUDA_VER=${CUDA_VER//./} && python3.10 -m pip install \
+RUN CUDA_VER=${CUDA_VERSION%.*} && CUDA_VER=${CUDA_VER//./} && python3.10 -m pip install --no-cache-dir \
     torch==2.0.1+cu${CUDA_VER} \
     torchvision==0.15.2+cu${CUDA_VER} \
-        --extra-index-url https://download.pytorch.org/whl/cu${CUDA_VER}
```
Comment (jb-ye) [inline Dockerfile:127]: fix indent

**Review by jb-ye — COMMENTED** [review]
Could you rebase the change and fix the indent? I think this change is helpful.

**Comment by bro-sin** [comment]
> Could you rebase the change and fix the indent? I think this change is helpful.

Thank you @jb-ye  for your advice. I have rebased my changes on top of the latest commit and fixed the indent issues.

**Comment by jb-ye** [comment]
LGTM

**Review by brentyi — APPROVED** [review]

**Comment by brentyi** [comment]
thanks @bro-sin @jb-ye!

**Pull request merged**